tests/torchtune/modules/loss/test_liger_ce_loss.py

```diff
@@ -0,0 +1,205 @@
+            dtype=torch.float32,
+            generator=gen,
+            device=device,
+        )
+
+        targets = torch.randint(
+            0,
+            VOCAB_SIZE,
+            (BATCH_SIZE, SEQ_LEN),
+            dtype=torch.long,
+            generator=gen,
+            device=device,
+        )
+
+        # Add ignored indices
+        mask = torch.rand((BATCH_SIZE, SEQ_LEN), generator=gen, device=device) < 0.2
+        targets[mask] = IGNORE_INDEX
+
+        # Test liger loss
+        loss_fn = LigerLinearCrossEntropy(ignore_index=IGNORE_INDEX)
+        loss_fn.set_model_output(fused_model)
+        if compile:
+            loss_fn.apply_compile_strategy()
+        fused_loss = self.loss_step(fused_model, loss_fn, embed, targets)
+
+        # Test standard loss
+        def standard_loss_fn(hidden, targets):
+            logits = model.output(hidden)
+            logits = logits.reshape(-1, VOCAB_SIZE)
+            targets = targets.reshape(-1)
+            loss = F.cross_entropy(
+                logits, targets, reduction="mean", ignore_index=IGNORE_INDEX
+            )
+            return loss
+
+        if compile:
+            torch.compile(standard_loss_fn)
+        standard_loss = self.loss_step(model, standard_loss_fn, embed, targets)
+
+        return fused_loss, standard_loss
+
+    @gpu_test(gpu_count=1)
+    @pytest.mark.parametrize("compile", [False, True])
+    def test_liger_fused_cross_entropy(self, compile):
+        """Test gradient flow through full forward/backward pass with optimizer step"""
+        # Create model for fused loss
+        fused_model = Model(VOCAB_SIZE, EMBED_DIM).cuda()
+        fixed_init_model(fused_model, min_val=-0.1, max_val=0.1)
+
+        # Create model for standard loss
+        model = Model(VOCAB_SIZE, EMBED_DIM).cuda()
+        fixed_init_model(model, min_val=-0.1, max_val=0.1)
+
+        fused_loss, standard_loss = self.compute_loss(
+            fused_model, model, "cuda", compile
+        )
+
+        # Verify:
+        # 1. Validate the results are close enough
+        assert_expected(fused_loss, standard_loss, rtol=1e-2, atol=1e-2)
+
+        # 2. Validate the weight updates are close enough
+        for p1, p2 in zip(fused_model.parameters(), model.parameters()):
+            assert_expected(p1, p2, rtol=1e-2, atol=1e-2)
+
+    def loss_worker(self, rank, compile, port):
+        os.environ["MASTER_ADDR"] = "localhost"
+        os.environ["MASTER_PORT"] = str(port)
+        device = torch.device(f"cuda:{rank}")
+        dist.init_process_group(
+            backend="nccl", rank=rank, world_size=WORLD_SIZE, device_id=device
+        )
+        torch.cuda.set_device(device)
+        mesh = ParallelDims(
+            dp_replicate=1, dp_shard=WORLD_SIZE // 2, tp=2, cp=1, world_size=WORLD_SIZE
+        ).build_mesh("cuda")
+        plan = {
+            "output": ColwiseParallel(
+                input_layouts=Replicate(), output_layouts=Replicate()
+            )
+        }
```

Comment:
Is this actually a nontrivial application of TP? (Like should we at least have some shard for `layer` or something?)